docs/groovy-script/mods/selectionguicrafting/recipe.md

```diff
@@ -0,0 +1,255 @@
+
+## Description
+
+Creates a recipe that is shown in the specified category. Each recipe requires at least an input (placed in the offhand), tool (placed in the mainhand), and output. There can also be an optional catalyst (placed in the inventory). The recipe can have its own frame, sounds, particles, and progress bar. It also allows to specify how the sounds are played, if the recipe can be added to the crafting queue, how the output items are handed to the player, and how much durability is consumed, if the tool is a damageable item. The crafting time as well as a possible XP reward can be set as well.
+
+## Identifier
+
+Refer to this via any of the following:
+
+```groovy:no-line-numbers {1}
+mods.selectionguicrafting.recipe/* Used as page default */ // [!code focus]
+mods.selectionguicrafting.Recipe
+```
+
+
+## Adding Recipes
+
+### Recipe Builder
+
+Just like other recipe types, the Crafting Recipe also uses a recipe builder.
+
+Don't know what a builder is? Check [the builder info page](../../getting_started/builder.md) out.
+
+:::::::::: details mods.selectionguicrafting.recipe.recipeBuilder() {open id="abstract"}
+- `Integer`. The amount of XP that will be granted to the player when the recipe is crafted. Requires greater than or equal to 0. (Default `0`).
+
+    ```groovy:no-line-numbers
+    xp(int)
+    ```
+
+- `Integer`. The time in ticks it takes to craft the recipe. Requires greater than or equal to 0. (Default `0`).
+
+    ```groovy:no-line-numbers
+    time(int)
+    ```
+
+- `ArrayList<GsTool>`. The tool that will be used to craft the recipe. Tool refers to the mainhand item. This doesn't need to be an item with durability. The first `float` value is the multiplier for the durability consumption. The second `float` value is the multiplier for the crafting time. Requires greater than or equal to 1.
+
+    ```groovy:no-line-numbers
+    tool(IIngredient)
+    tool(IIngredient, float)
+    tool(IIngredient, float, float)
+    ```
+
+- `ResourceLocation`. The path to the frame texture. Overrides the category frame. [Textures that come with the mod](https://github.com/Ender-Development/selection-gui-crafting-continued/tree/master/src/main/resources/assets/selectionguicrafting/textures/gui/frame). (Default `selectionguicrafting:textures/gui/frame/default.png`).
+
+    ```groovy:no-line-numbers
+    frame(String)
+    frame(ResourceLocation)
+    ```
+
+- `ArrayList<Ingredient>`. The input items that are required to craft the recipe. Input refers to the offhand item. You can add multiple inputs each with a different amount. Requires greater than or equal to 1.
+
+    ```groovy:no-line-numbers
+    input(IIngredient)
+    ```
+
+- `ArrayList<GsOutput>`. The output items that will be given to the player. You can add multiple outputs each with a different chance. The `float` value is the chance of the item being dropped. Requires greater than or equal to 1.
+
+    ```groovy:no-line-numbers
+    output(ItemStack)
```

Comment:
same as `input` (`Collection<ItemStack>`, `ItemStack...`) but also should include methods to accept `GsOutput` directly (and, collection/varargs of it)